Stratis.Bitcoin.Features.Consensus/StakeItem.cs

```diff
@@ -1,13 +1,84 @@
 ﻿using NBitcoin;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.Features.Consensus
 {
-    public class StakeItem
+    /// <summary>
+    /// An object that holds block stake information.
+    /// </summary>
+    public sealed class StakeItem
     {
-        public uint256 BlockId;
-        public BlockStake BlockStake;
-        public bool InStore;
-        public long Height;
-    }
-}
+        private StakeItem() { }
+
+        /// <summary>The hash of the block.</summary>
+        public uint256 BlockHash { get; private set; }
+
+        /// <summary>The block stake.</summary>
+        public BlockStake BlockStake { get; private set; }
+
+        /// <summary>The height of the block.</summary>
+        public long? Height { get; private set; }
+
+        /// <summary>
+        /// Returns <c>true</c> if the item exists in the store. It is used to determine
+        /// if the <see cref="StakeChainStore"/> should persist the item or not. />
+        /// </summary>
+        public bool ExistsInStore { get; private set; }
+
+        /// <summary>
+        /// Sets the item's <see cref="ExistsInStore"/> value to <c>true</c>.
+        /// </summary>
+        internal void InStore()
+        {
+            this.ExistsInStore = true;
+        }
+
+        /// <summary>
+        /// Sets the item's <see cref="BlockStake"/> value.
+        /// </summary>
+        internal void Update(BlockStake blockStake)
+        {
+            Guard.NotNull(blockStake, nameof(blockStake));
 
+            this.BlockStake = blockStake;
+
+            InStore();
+        }
+
+        /// <summary>
+        /// Static method to create a valid <see cref="StakeItem"/> object.
```

Comment:
I don't think it is beneficial to repeat here that it is static method. I'd just use `Creates a valid <see cref="StakeItem"/> object.`